File: TopflytechCodec/Java/topflytechCodec/src/main/java/com/topflytech/codec/entities/Message.java
package com.topflytech.codec.entities;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import java.util.Date;

/**
 * Message is the base class for all decoded messages
 */
public abstract class Message {
    /**
     * Gets imei.
     *
     * @return the imei
     */
    public String getImei() {
        return imei;
    }

    /**
     * Sets imei.
     *
     * @param imei the imei
     */
    public void setImei(String imei) {
        this.imei = imei;
    }

    /**
     * Gets serial number of the message,The serial number is counted on the device
     *
     * @return the serial no
     */
    public int getSerialNo() {
        return serialNo;
    }

    /**
     * Sets serial number.
     *
     * @param serialNo the serial no
     */
    public void setSerialNo(int serialNo) {
        this.serialNo = serialNo;
    }

    /**
     * Get original bytes byte [ ].
     *
     * @return the byte [ ]
     */
    public byte[] getOrignBytes() {
        return orignBytes;
    }

    /**
     * Sets original bytes.
     *
     * @param orignBytes the orign bytes
     */
    public void setOrignBytes(byte[] orignBytes) {
        this.orignBytes = orignBytes;
    }
    public boolean isNeedResp() {
        return isNeedResp;
    }

    public void setIsNeedResp(boolean isNeedResp) {
        this.isNeedResp = isNeedResp;
    }
    public int getProtocolHeadType() {
        return protocolHeadType;
    }

    public void setProtocolHeadType(int protocolHeadType) {
        this.protocolHeadType = protocolHeadType;
    }

    public int getEncryptType() {
        return encryptType;
    }

    public void setEncryptType(int encryptType) {
        this.encryptType = encryptType;
    }
  
    private String imei;
    private int serialNo;
    private byte[] orignBytes;
    private boolean isNeedResp = true;
    private int protocolHeadType;
    private int encryptType = MessageEncryptType.NONE;


    //The following are used for other purposes.
     /**
     * Convert the message to JSON string
     * @return JSON string representation of the message
     */
    public String toJSON() {
        try { 
            return JSON.toJSONString(this);
        } catch (Exception e) { 
            return "{}";
        }
    }
      private String protocol;
    private String linkType; 

    private Date recvDate;
    private String postResp;

     public void setProtocol(String protocol) {
        this.protocol = protocol;
    }
    public String getProtocol() {
        return protocol;
    }

    public String getLinkType() {
		return linkType;
	}

    public void setLinkType(String linkType) {
		this.linkType = linkType;
	}
    public Date getRecvDate() {
        return recvDate;
    }
    public void setRecvDate(Date recvDate) {
        this.recvDate = recvDate;
    }
    public String getPostResp() {
        return postResp;
    }
    public void setPostResp(String postResp) {
        this.postResp = postResp;
    }
}

File: TopflytechCodec/C#/TopflytechCodec/TopflytechCodec/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TopflytechCodec.Entities
{
    /// <summary>
    /// Message is the base class for all decoded messages
    /// </summary>
    public abstract class Message
    {
        private String imei;
        public System.String Imei
        {
            get { return imei; }
            set { imei = value; }
        }
        private int serialNo;
        /// <summary>
        /// The serial number of the message,The serial number is counted on the device
        /// </summary>
        public int SerialNo
        {
            get { return serialNo; }
            set { serialNo = value; }
        }
        private byte[] orignBytes; 
        /// <summary>
        /// The original bytes.
        /// </summary>
        public byte[] OrignBytes
        {
            get { return orignBytes; }
            set { orignBytes = value; }
        }
        private Boolean isNeedResp = true;
        public Boolean IsNeedResp
        {
            get { return isNeedResp; }
            set { isNeedResp = value; }
        }

        private int protocolHeadType;
        public int ProtocolHeadType
        {
            get { return protocolHeadType; }
            set { protocolHeadType = value; }
        }

        private int encryptType;
        public int EncryptType
        {
            get { return encryptType; }
            set { encryptType = value; }
        }

        private String protocol;
        public String Protocol
        {
            get { return protocol; }
            set { protocol = value; }
        }

        private String linkType;
        public String LinkType
        {
            get { return linkType; }
            set { linkType = value; }
        }

        private DateTime recvDate;
        public DateTime RecvDate
        {
            get { return recvDate; }
            set { recvDate = value; }
        }

        private String postResp;
        public String PostResp
        {
            get { return postResp; }
            set { postResp = value; }
        }
    }
}
